### PR TITLE
Task-38524: Display external label for tasks

### PR DESCRIPTION
**Pull request by ahmadmido**

### CODE REVIEW

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskAssignment.vue [hunk 1]
```diff
@@ -21,7 +21,7 @@
             class="assigneeName">
             <exo-user-avatar
               :username="taskAssigneeObj.profile.remoteId"
-              :fullname="taskAssigneeObj.profile.fullName"
+              :fullname="fullName"
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskAssignment.vue]:
```suggestion
              :fullname="taskAssigneeFullName"
```

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskAssignment.vue [hunk 2]
```diff
@@ -161,6 +164,7 @@
                 profile: {
                   fullName: user.profile.fullname,
                   avatarUrl: user.profile.avatar,
+                  external: user.profile.dataEntity.external !== null && user.profile.dataEntity.external === 'true',
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskAssignment.vue]: Do you need to test user.profile.dataEntity.external !== null ?

File: task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue [hunk 3]
```diff
@@ -38,8 +43,17 @@ export default {
     },
     displayName() {
       const profile = this.attendee && (this.attendee.profile || this.attendee.space);
-      return profile && (profile.displayName || profile.fullname || profile.fullName);
+      const fullName = profile && (profile.displayName || profile.fullname || profile.fullName);
+      return this.isExternal && fullName.indexOf(this.$t('label.external')) < 0 ? fullName.concat(' (').concat(this.$t('label.external')).concat(')') : fullName;
     },
   },
+  created() {
+   if(this.attendee.providerId === 'organization'){
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue:51]: Why doing this test ?

File: task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue [hunk 4]
```diff
@@ -38,8 +43,17 @@ export default {
     },
     displayName() {
       const profile = this.attendee && (this.attendee.profile || this.attendee.space);
-      return profile && (profile.displayName || profile.fullname || profile.fullName);
+      const fullName = profile && (profile.displayName || profile.fullname || profile.fullName);
+      return this.isExternal && fullName.indexOf(this.$t('label.external')) < 0 ? fullName.concat(' (').concat(this.$t('label.external')).concat(')') : fullName;
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue]: Why do you need this test: fullName.indexOf(this.$t('label.external')) < 0

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue (hunk 3 above)
Comment (ahmadmido) [reply task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue:51]: To assure that we don't call the space providerId

File: task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue (hunk 4 above)
Comment (ahmadmido) [reply task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue]: This test to assure that we have a unique external tag when we display users after selections

Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectEventFormAssigneeAndParticipatorItem.vue]: Please remove concat

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Review by rdenarie — APPROVED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**